projects/lpc40xx_freertos/l5_application/LAB_04/line_buffer.c

```diff
@@ -0,0 +1,66 @@
+#include "line_buffer.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+void line_buffer__init(line_buffer_s *buffer, void *memory, size_t size) {
+  if (buffer != NULL) {
+    buffer->memory = memory;
+    buffer->max_size = size;
+    buffer->write_index = 0;
```

Comment:
`memset(buffer->memory, 0, buffer->max_size)`

you can reset the buffer memory in init

Reply:
Ohh ya.. Didn't occur to me at that time.. Thanks!